### PR TITLE
A generate_docs command

### PR DESCRIPTION
**Pull request by nedbat**
**Description:** Describe in a couple of sentence what this PR adds

**JIRA:** Link to JIRA ticket

**Dependencies:** dependencies on other outstanding PRs, issues, etc. 

**Merge deadline:** List merge deadline (if any)

**Installation instructions:** List any non-trivial installation 
instructions.

**Testing instructions:**

1. Do thing A
2. Expect B to happen
3. If C happened instead - check failed.

**Reviewers:**
- [ ] tag reviewer 
- [ ] tag reviewer 

**Merge checklist:**
- [ ] All reviewers approved
- [ ] CI build is green
- [ ] Version bumped
- [ ] Changelog record added
- [ ] Documentation updated (not only docstrings)
- [ ] Commits are squashed
- [ ] PR author is listed in AUTHORS

**Post merge:**
- [ ] Create a tag
- [ ] Check new version is pushed to PyPi after tag-triggered build is 
      finished.
- [ ] Delete working branch (if not needed anymore)

**Author concerns:** List any concerns about this PR - inelegant 
solutions, hacks, quick-and-dirty implementations, concerns about 
migrations, etc.

### CODE REVIEW

**Comment by nedbat** [comment]
There are no tests yet, but the existing tests still pass.

**Comment by nedbat** [comment]
Also, note this is on top of @bmedx's grouping PR

File: code_annotations/base.py [hunk 1]
```diff
@@ -69,6 +69,8 @@ def __init__(self, config_file_path, report_path_override, verbosity, source_pat
         self.source_path = source_path_override if source_path_override else raw_config['source_path']
         self.echo("Configured for source path: {}".format(self.source_path))
 
+        self.rst_template = raw_config.get('rst_template')
```
Comment (bmedx) [inline code_annotations/base.py]: Can we get a default/sample template added with this to show how it should look?

File: code_annotations/cli.py [hunk 2]
```diff
@@ -171,3 +166,49 @@ def static_find_annotations(config_file, source_path, report_path, verbosity, li
...
+
+
+@entry_point.command("generate_docs")
+@click.option(
+    '--config_file',
+    default='.annotations',
+    help='Path to the configuration file',
+    type=click.Path(exists=True, dir_okay=False, resolve_path=True)
+)
+@click.argument("report", type=click.File('r'))
```
Comment (bmedx) [inline code_annotations/cli.py]: Eventually we were hoping to be able to feed in multiple report files and have them rolled together to gather output from the various tools (or various repos) together in the doc. Not a blocker for this, but if you find a minute to add it that would be grand.

**Pull request closed without merging**